interfaces/mount/entry.go

```diff
@@ -0,0 +1,105 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package mount
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Entry describes an /etc/fstab-like mount entry.
+//
+// Fields are named after names in struct returned by getmntent(3).
+//
+// struct Mntent {
+//     char *Mnt_fsname;   /* name of mounted filesystem */
+//     char *Mnt_dir;      /* filesystem path prefix */
+//     char *Mnt_type;     /* mount type (see Mntent.h) */
+//     char *Mnt_opts;     /* mount options (see Mntent.h) */
+//     int   Mnt_freq;     /* dump frequency in days */
+//     int   Mnt_passno;   /* pass number on parallel fsck */
+// };
+type Entry struct {
+	FsName  MntFsName
+	Dir     MntDir
+	FsType  MntFsType
+	Options MntOptions
```

Comment:
You are right. I'll go ahead with non-legacy names (go did this a while ago and I think it was regarded as a mistake). As for the over-typing I think it is fine to use strings as long as the String() function does the special casing of empty values (which is easy).

Thanks for the feedback. I'll tweak this and push again.